editoast/src/chartos/layer_cache.rs

```diff
@@ -0,0 +1,103 @@
+use core::f64::consts::PI;
+
+use super::BoundingBox;
+
+// File containing dead_code until https://github.com/DGEXSolutions/osrd/issues/1795 is done
+
+#[allow(dead_code)]
+pub struct Tile {
+    x: i64,
+    y: i64,
+    z: i64,
+}
+
+#[allow(dead_code)]
+fn xy_from_latitude_longitude(latitude: f64, longitude: f64, zoom: i64) -> (i64, i64) {
+    let n = 2.0_f64.powf(zoom as f64);
+    (
+        ((longitude + 180.0) / 360.0 * n).floor() as i64,
+        ((1.0 - latitude.to_radians().tan().asinh() / PI) / 2.0 * n).floor() as i64,
+    )
+}
+
+#[allow(dead_code)]
+fn get_nw_se_coordinates(zoom: i64, bbox: &BoundingBox) -> (i64, i64, i64, i64) {
+    assert!(bbox.is_valid());
```

Comment:
I'm not sure if it should panic here or not

Reply:
This seems normal. You can specify this in the documentation of the function.

Reply:
Can you specify that `bbox` needs to be valid otherwise the function panics.